tests/components/modbus/conftest.py

```diff
@@ -67,8 +68,11 @@ async def base_test(
         "homeassistant.components.modbus.modbus.ModbusUdpClient", return_value=mock_sync
     ):
 
-        # Setup inputs for the sensor
-        read_result = ReadResult(register_words)
+        read_result = (
+            register_words_or_exception
+            if isinstance(register_words_or_exception, Exception)
+            else ReadResult(register_words_or_exception)
+        )
```

Comment:
This is a good idea, but we do not want to test pymodbus in every platform, that is the responsibiliy and thus testing of modbus.py.

Reply:
It's not a Modbus test, it's a negative case testing wish Modbus integration lack of, unfortuately